pkg/image/sif.go

```diff
@@ -38,6 +40,12 @@ func (f *sifFormat) initializer(img *Image, fileinfo os.FileInfo) error {
 		return err
 	}
 
+	// Check the compatibility of the image's target architecture
+	sifArch := string(fimg.Header.Arch[:sif.HdrArchLen-1])
+	if sifArch != sif.HdrArchUnknown && sifArch != sif.GetSIFArch(runtime.GOARCH) {
```

Comment:
like you said it's good for now. Comment for reminder on architecture like i386 can run on amd64, maybe some arm processor can run <= armv6 instructions on aarch64, however not 100% sure.

Reply:
Good idea and done. I tried to capture the entire situation and hopefully will work on better support during the next few weeks. Meanwhile I believe this PR is ready to go.